src/tableutils.js

```diff
@@ -0,0 +1,398 @@
+	 */
+	createTable( position, rows, columns ) {
+		const model = this.editor.model;
+
+		model.change( writer => {
+			const table = writer.createElement( 'table' );
+
+			writer.insert( table, position );
+
+			createEmptyRows( writer, table, 0, rows, columns );
+		} );
+	}
+
+	/**
+	 * Insert rows into a table.
+	 *
+	 * @param {module:engine/model/element~Element} table
+	 * @param {Object} options
+	 * @param {Number} [options.at=0] Row index at which insert rows.
+	 * @param {Number} [options.rows=1] Number of rows to insert.
+	 */
+	insertRows( table, options = {} ) {
+		const model = this.editor.model;
+
+		const insertAt = options.at || 0;
+		const rows = options.rows || 1;
+
+		const headingRows = table.getAttribute( 'headingRows' ) || 0;
+
+		model.change( writer => {
+			if ( headingRows > insertAt ) {
+				writer.setAttribute( 'headingRows', headingRows + rows, table );
+			}
+
+			const tableIterator = new TableWalker( table, { endRow: insertAt + 1 } );
+
+			let tableCellToInsert = 0;
+
+			for ( const tableCellInfo of tableIterator ) {
+				const { row, rowspan, colspan, cell } = tableCellInfo;
+
+				const isBeforeInsertedRow = row < insertAt;
+				const overlapsInsertedRow = row + rowspan > insertAt;
+
+				if ( isBeforeInsertedRow && overlapsInsertedRow ) {
+					writer.setAttribute( 'rowspan', rowspan + rows, cell );
+				}
+
+				// Calculate how many cells to insert based on the width of cells in a row at insert position.
+				// It might be lower then table width as some cells might overlaps inserted row.
+				if ( row === insertAt ) {
+					tableCellToInsert += colspan;
+				}
+			}
+
+			// If insertion occurs on the end of a table use table width.
+			if ( insertAt >= table.childCount ) {
+				tableCellToInsert = this.getColumns( table );
+			}
+
+			createEmptyRows( writer, table, insertAt, rows, tableCellToInsert );
+		} );
+	}
+
+	/**
+	 * Inserts columns into a table.
+	 *
+	 * @param {module:engine/model/element~Element} table
+	 * @param {Object} options
+	 * @param {Number} [options.at=0] Column index at which insert columns.
+	 * @param {Number} [options.columns=1] Number of columns to insert.
+	 */
+	insertColumns( table, options = {} ) {
+		const model = this.editor.model;
+
+		const insertAt = options.at || 0;
+		const columns = options.columns || 1;
+
+		model.change( writer => {
+			const tableColumns = this.getColumns( table );
+
+			// Inserting at the end and at the begging of a table doesn't require to calculate anything special.
+			if ( insertAt === 0 || tableColumns <= insertAt ) {
+				for ( const tableRow of table.getChildren() ) {
+					createCells( columns, writer, Position.createAt( tableRow, insertAt ? 'end' : 0 ) );
+				}
+
+				return;
+			}
+
+			const headingColumns = table.getAttribute( 'headingColumns' );
+
+			if ( insertAt < headingColumns ) {
+				writer.setAttribute( 'headingColumns', headingColumns + columns, table );
+			}
+
+			const tableMap = [ ...new TableWalker( table ) ];
```

Comment:
It might be usefull while doing some things on columns. The only thing to check is how make this usefull - including cells that spans over current column or not? I'll go through the usages though.